scripts/zkai_attention_kv_model_faithful_quantized_attention_bridge_gate.py

```
@@ -0,0 +1,640 @@
+        if denominator <= 0:
+            raise ModelFaithfulQuantizedAttentionBridgeGateError("model policy produced nonpositive denominator")
+        per_step_denominators.append(denominator)
+
+        numerators = [0 for _ in range(value_width)]
+        for (candidate, _score), weight in zip(scored, weights, strict=True):
+            for dim, value in enumerate(candidate["value"]):
+                numerators[dim] += weight * value
+
+        output: list[int] = []
+        remainders: list[int] = []
+        for dim, numerator in enumerate(numerators):
```

Ruff (0.15.12)

[warning] Loop control variable `dim` not used within loop body

(B007)